packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -757,6 +727,96 @@ class IOSDevice extends Device {
     _logger.printError('');
   }
 
+  /// Find the Dart VM url using ProtocolDiscovery (logs from `idevicesyslog`)
+  /// and mDNS simultaneously, using whichever is found first. `idevicesyslog`
+  /// does not work on wireless devices, so only use mDNS for wireless devices.
+  /// Wireless devices require using the device IP as the host.
+  Future<Uri?> _discoverDartVMForCoreDevice({
+    required String packageId,
+    required bool ipv6,
+    required DebuggingOptions debuggingOptions,
+    ProtocolDiscovery? vmServiceDiscovery,
+  }) async {
+    Timer? maxWaitForCI;
+    final Completer<Uri?> cancelCompleter = Completer<Uri?>();
+
+    // When testing in CI, wait a max of 10 minutes for the Dart VM to be found.
+    // Afterwards, stop the app from running and upload DerivedData Logs to debug
+    // logs directory. CoreDevices are run through Xcode and launch logs are
+    // therefore found in DerivedData.
+    if (debuggingOptions.usingCISystem && debuggingOptions.debugLogsDirectoryPath != null) {
+      maxWaitForCI = Timer(const Duration(minutes: 10), () async {
```

Comment:
I wonder, will we get timed out before 10 minutes pass? Does it need to be this long? I'm not sure how long this could reasonably take...

Reply:
in looking at a few logs, 10 minutes seems fine